panther_manager/include/panther_manager/manager_bt_node.hpp

```diff
@@ -9,30 +10,41 @@
 #include <ros/ros.h>
 
 #include <std_msgs/Bool.h>
+#include <sensor_msgs/BatteryState.h>
 
+#include <panther_msgs/DriverState.h>
 #include <panther_msgs/IOState.h>
+#include <panther_msgs/SystemStatus.h>
+
+#include <panther_manager/moving_average.hpp>
 
 namespace panther_manager
 {
 
 class ManagerNode
 {
 public:
-  ManagerNode(std::shared_ptr<ros::NodeHandle> nh);
+  ManagerNode(std::shared_ptr<ros::NodeHandle> nh, std::shared_ptr<ros::NodeHandle> pnh);
   ~ManagerNode() {}
 
 private:
+  float battery_percent_;
+  float update_charging_anim_step_;
+  float shutdown_timeout_;
   std::string node_name_;
-  std::string xml_filename_;
+  std::optional<unsigned> battery_status_;
```

Comment:
```suggestion
  std::optional<std::uint8_t> battery_status_;
```
The message field is not purely unsigned

Reply:
After discussion I am dropping this idea